art/estimators/speech_recognition/pytorch_deep_speech.py

```diff
@@ -676,6 +724,43 @@ def _transform_model_input(
 
         return inputs, targets, input_percentages, target_sizes, batch_idx
 
+    def get_transformation_params(self) -> Tuple[int, str, int, int, int]:
+        """
+        Get parameters needed for audio transformation.
+
+        :return: A tuple of (sample_rate, window_name, win_length, n_fft, hop_length)
+                    - sample_rate: audio sampling rate.
+                    - window_name: the type of window to create.
+                    - win_length: the number of samples in the window.
+                    - n_fft: FFT window size.
+                    - hop_length: number audio of frames between STFT columns.
+        """
+        # These parameters are needed for audio transformation
+        window_name = self.model.audio_conf.window.value
+        sample_rate = self.model.audio_conf.sample_rate
+        window_size = self.model.audio_conf.window_size
+        window_stride = self.model.audio_conf.window_stride
+
+        n_fft = int(sample_rate * window_size)
+        hop_length = int(sample_rate * window_stride)
+        win_length = n_fft
+
+        return sample_rate, window_name, win_length, n_fft, hop_length
+
+    def to_training_mode(self) -> None:
+        """
+        Put the estimator in the training mode.
+        """
+        self.model.train()
+
+    def batch_norm(self, train: bool) -> None:
+        """
+        Set all batch normalization layers into train or eval mode.
+
+        :param train: False for evaluation mode.
+        """
+        self.set_batchnorm(train=train)
```

Comment:
What is the reason for this method? It just seems to call the existing method.

Reply:
@beat-buesser : We need to call this in the attack for the Deepspeech estimator, but the Espresso one does not need to call set_batchnorm, so they need to provide an empty method. Please note that the set_batchnorm is inherited from the PytorchEstimator where Espresso also inherits, so we cannot call this method directly.